scrapers_next/al/people.py

```diff
@@ -0,0 +1,144 @@
+
+        table = CSS("#ContentPlaceHolder1_TabSenator_TabLeg_gvLEG").match_one(self.root)
+
+        for tr in CSS("tr").match(table):
+            type, info = CSS("td").match(tr)
+            type = type.text_content()
+            info = info.text_content()
+
+            if type == "Affiliation:":
+                party = ""
+                if info == "(R)":
+                    party = "Republican"
+                elif info == "(D)":
+                    party = "Democrat"
+                else:
+                    party = info
+                    if party == "( )":
+                        self.logger.warning("this seat may be vacant")
+                        # will assign it a random party for scraping purposes, but the warning should be addressed:
+                        party = "Democrat"
+
+            elif type == "District:":
+                district = info.split(" ")[2]
+            elif type == "County:":
+                county = info
+            elif type == "Phone Number:":
+                phone = info
+            elif type == "Fax Number:":
+                if info != "":
+                    fax = info
+            elif type == "Street:":
+                street = info
+            elif type == "Office:":
+                office = info
+            elif type == "City:":
+                city = info
+            elif type == "Postal Code:":
+                postal = info
+            elif type == "Email:":
+                email = info
+
+        address = f"{street}, {office}, {city} AL"
+
+        image = (
+            CSS("#ContentPlaceHolder1_TabSenator_TabLeg_imgLEG")
+            .match_one(self.root)
+            .get("src")
+        )
+
+        p = ScrapePerson(
+            name=full_name.title(),
+            state="al",
+            chamber=self.input.chamber,
+            party=party,
+            district=district,
+            email=email,
+            image=image,
+        )
+        p.add_source(self.source.url)
+        p.add_source(self.input.url)
+
+        # This address is the capitol office
+        if re.search("11 South Union Street", street):
+            p.capitol_office.address = address
+            p.capitol_office.voice = phone
+            try:
+                p.capitol_office.fax = fax
+            except ValueError:
+                pass
+        else:
+            p.district_office.address = address
+            p.district_office.voice = phone
+            try:
+                p.district_office.fax = fax
+            except ValueError:
+                pass
+
+        p.extras["postal code"] = postal
+        p.extras["county"] = county
+
+        return p
+
+
+class LegList(HtmlListPage):
+    def process_item(self, item):
+        last_name = re.split("Pictures/|_", item.get("src"))[1]
+        oid_person = item.get("alt")
+
+        # specific numbers in the img tag make up parts of the URL
+
+        if self.chamber == "upper":
```

Comment:
generally it is best practice to not have code in a parent class that depends on knowing things from the child class

applied here that'd mean that instead of having one process_item function, you'd have two (one on RepList and one on SenList) and put the appropriate code there instead of in this conditional